drivers/flash/nrf_qspi_nor.c

```diff
@@ -0,0 +1,690 @@
+		return NRF_QSPI_FREQ_32MDIV14;  /**< 2.29 MHz. */
+	} else if (frequency < 2660000) {
+		return NRF_QSPI_FREQ_32MDIV13;  /**< 2.46 MHz. */
+	} else if (frequency < 2900000) {
+		return NRF_QSPI_FREQ_32MDIV12;  /**< 2.66 MHz. */
+	} else if (frequency < 3200000) {
+		return NRF_QSPI_FREQ_32MDIV11;  /**< 2.90 MHz. */
+	} else if (frequency < 3550000) {
+		return NRF_QSPI_FREQ_32MDIV10;  /**< 3.20 MHz. */
+	} else if (frequency < 4000000) {
+		return NRF_QSPI_FREQ_32MDIV9;   /**< 3.55 MHz. */
+	} else if (frequency < 4570000) {
+		return NRF_QSPI_FREQ_32MDIV8;   /**< 4.00 MHz. */
+	} else if (frequency < 5330000) {
+		return NRF_QSPI_FREQ_32MDIV7;   /**< 4.57 MHz. */
+	} else if (frequency < 6400000) {
+		return NRF_QSPI_FREQ_32MDIV6;   /**< 5.33 MHz. */
+	} else if (frequency < 8000000) {
+		return NRF_QSPI_FREQ_32MDIV5;   /**< 6.40 MHz. */
+	} else if (frequency < 10600000) {
+		return NRF_QSPI_FREQ_32MDIV4;   /**< 8.00 MHz. */
+	} else if (frequency < 16000000) {
+		return NRF_QSPI_FREQ_32MDIV3;   /**< 10.6 MHz. */
+	} else if (frequency < 32000000) {
+		return NRF_QSPI_FREQ_32MDIV2;   /**< 16.0 MHz. */
+	} else {
+		return NRF_QSPI_FREQ_32MDIV1;   /**< 32.0 MHz. */
+	}
+}
+
+static inline nrf_qspi_addrmode_t qspi_get_address_size(bool addr_size)
+{
+	return addr_size ? NRF_QSPI_ADDRMODE_32BIT : NRF_QSPI_ADDRMODE_24BIT;
+}
+
+/**
+ * @brief Test whether offset is aligned.
+ */
+#define QSPI_IS_SECTOR_ALIGNED(_ofs) (((_ofs) & (QSPI_SECTOR_SIZE - 1U)) == 0)
+#define QSPI_IS_BLOCK_ALIGNED(_ofs) (((_ofs) & (QSPI_BLOCK_SIZE - 1U)) == 0)
+
+/**
+ * @brief Main configuration structure
+ */
+static struct qspi_nor_data qspi_nor_memory_data = {
+	.sem = Z_SEM_INITIALIZER(qspi_nor_memory_data.sem, 1, 1),
+	.sync = Z_SEM_INITIALIZER(qspi_nor_memory_data.sync, 0, 1),
+};
+
+/**
+ * @brief Converts NRFX return codes to the zephyr ones
+ */
+static inline int qspi_get_zephyr_ret_code(int res)
+{
+	switch (res) {
+	case NRFX_SUCCESS:
+		return 0;
+	case NRFX_ERROR_INVALID_PARAM:
+	case NRFX_ERROR_INVALID_ADDR:
+		return -EINVAL;
+	case NRFX_ERROR_INVALID_STATE:
+		return -ECANCELED;
+	case NRFX_ERROR_BUSY:
+	case NRFX_ERROR_TIMEOUT:
+	default:
+		return -EBUSY;
+	}
+}
+
+static inline struct qspi_nor_data *get_dev_data(struct device *dev)
+{
+	return dev->driver_data;
+}
+
+static inline void qspi_lock(struct device *dev)
+{
+	struct qspi_nor_data *dev_data = get_dev_data(dev);
+
+	k_sem_take(&dev_data->sem, K_FOREVER);
+}
+
+static inline void qspi_unlock(struct device *dev)
+{
+	struct qspi_nor_data *dev_data = get_dev_data(dev);
+
+	k_sem_give(&dev_data->sem);
+}
+
+static inline void qspi_wait_for_completion(struct device *dev)
+{
+	struct qspi_nor_data *dev_data = get_dev_data(dev);
+
+	k_sem_take(&dev_data->sync, K_FOREVER);
+}
+
+static inline void qspi_complete(struct device *dev)
+{
+	struct qspi_nor_data *dev_data = get_dev_data(dev);
+
+	k_sem_give(&dev_data->sync);
+}
+
+/**
+ * @brief QSPI handler
+ *
+ * @param event - driver event type
```

Comment:
remove dash.
in other places of this file is saw colon after a perimeter line, I prefer consistency like `@param param_name what_param_does`